static/app/components/events/interfaces/crons/cronTimelineSection.tsx

```diff
@@ -0,0 +1,161 @@
+import {useRef} from 'react';
+import styled from '@emotion/styled';
+
+import {EventDataSection} from 'sentry/components/events/eventDataSection';
+import {Overlay} from 'sentry/components/overlay';
+import Panel from 'sentry/components/panels/panel';
+import Placeholder from 'sentry/components/placeholder';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {Event, Organization} from 'sentry/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import {useDimensions} from 'sentry/utils/useDimensions';
+import useRouter from 'sentry/utils/useRouter';
+import {CheckInTimeline} from 'sentry/views/monitors/components/overviewTimeline/checkInTimeline';
+import {
+  GridLineOverlay,
+  GridLineTimeLabels,
+} from 'sentry/views/monitors/components/overviewTimeline/gridLines';
+import {ResolutionSelector} from 'sentry/views/monitors/components/overviewTimeline/resolutionSelector';
+import {
+  MonitorBucketData,
+  TimeWindow,
+} from 'sentry/views/monitors/components/overviewTimeline/types';
+import {timeWindowConfig} from 'sentry/views/monitors/components/overviewTimeline/utils';
+import {getTimeRangeFromEvent} from 'sentry/views/monitors/utils/getTimeRangeFromEvent';
+
+interface Props {
+  event: Event;
+  organization: Organization;
+}
+
+const DEFAULT_ENVIRONMENT = 'production';
+
+export function CronTimelineSection({event, organization}: Props) {
+  const {location} = useRouter();
+  const timeWindow: TimeWindow = location.query?.timeWindow ?? '24h';
+  const monitorSlug = event.tags.find(({key}) => key === 'monitor.slug')?.value;
+  const environment = event.tags.find(({key}) => key === 'environment')?.value;
+
+  const nowRef = useRef<Date>(new Date());
+  const {start, end} = getTimeRangeFromEvent(event, nowRef.current, timeWindow);
+  const {elementRef, width: timelineWidth} = useDimensions<HTMLDivElement>();
+
+  const elapsedMinutes = timeWindowConfig[timeWindow].elapsedMinutes;
+  const rollup = Math.floor((elapsedMinutes * 60) / timelineWidth);
+
+  const monitorStatsQueryKey = `/organizations/${organization.slug}/monitors-stats/`;
+  const {data: monitorStats, isLoading} = useApiQuery<Record<string, MonitorBucketData>>(
+    [
+      monitorStatsQueryKey,
+      {
+        query: {
+          until: Math.floor(end.getTime() / 1000),
+          since: Math.floor(start.getTime() / 1000),
+          monitor: monitorSlug,
+          resolution: `${rollup}s`,
+        },
+      },
+    ],
+    {
+      staleTime: 0,
+      enabled: !!monitorSlug && timelineWidth > 0,
+    }
+  );
+
+  if (!monitorSlug) {
+    return null;
```

Comment:
Do we want to show a loading indicator w/ what @vuluongj20 built?

Reply:
oh yes definitely let me get that